src/main/java/de/tum/in/www1/artemis/web/rest/competency/PrerequisiteResource.java

```diff
@@ -0,0 +1,114 @@
+package de.tum.in.www1.artemis.web.rest.competency;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.PrerequisiteRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.security.annotations.EnforceAtLeastStudent;
+import de.tum.in.www1.artemis.security.annotations.enforceRoleInCourse.EnforceAtLeastEditorInCourse;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.service.competency.PrerequisiteService;
+import de.tum.in.www1.artemis.web.rest.dto.competency.PrerequisiteResponseDTO;
+
+/**
+ * REST controller for managing {@link de.tum.in.www1.artemis.domain.competency.Prerequisite Prerequisite} entities.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/")
+public class PrerequisiteResource {
+
+    private static final Logger log = LoggerFactory.getLogger(PrerequisiteResource.class);
+
+    private final PrerequisiteService prerequisiteService;
+
+    private final PrerequisiteRepository prerequisiteRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    public PrerequisiteResource(PrerequisiteService prerequisiteService, PrerequisiteRepository prerequisiteRepository, CourseRepository courseRepository,
+            AuthorizationCheckService authorizationCheckService) {
+        this.prerequisiteService = prerequisiteService;
+        this.prerequisiteRepository = prerequisiteRepository;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    /**
+     * GET /courses/:courseId/competencies/prerequisites
+     *
+     * @param courseId the id of the course for which the competencies should be fetched
+     * @return the ResponseEntity with status 200 (OK) and with body the found competencies
+     */
+    @GetMapping("courses/{courseId}/competencies/prerequisites")
+    @EnforceAtLeastStudent
```

Comment:
```suggestion
    @EnforceAtLeastStudentInCourse
```

Reply:
This is actually intended. We want to allow students to view the prerequisites of courses they can enroll into.
 If you look at the code below:
```
if (!course.isEnrollmentEnabled()) {
    authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
}
```
We only want to check if the student is in this course **if the self enrollment is not enabled**.

Reply:
I am not sure how this works: If one was not enrolled in a single course, would one be a student? What I want to say is: If one was enrolling in a course for the first time, would `@EnforceAtLeastStudent` permit them to see the prerequisites? 

If one has the role `user` until they first enrolled in a course, this could be an edge case to consider. If the role `student` is given to anyone who signed in to Artemis, this should not be an issue.

Reply:
Good point @Jan-Thurner . I looked up the annotation, it checks against the role "user" so this should be no problem (see `EnforceAtLeastStudent.java`)

![image](https://github.com/ls1intum/Artemis/assets/118574504/891ab14e-966c-4210-8007-2b0f7d286c6b)